impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -498,7 +498,22 @@ static std::optional<Entity> PipelineBlend(
   auto dst_snapshot =
       inputs[0]->GetSnapshot("PipelineBlend(Dst)", renderer, entity);
   if (!dst_snapshot.has_value()) {
-    return std::nullopt;
+    return std::nullopt;  // Nothing to render.
+  }
+
+  Rect subpass_coverage = coverage;
+  if (entity.GetContents()) {
+    auto coverage_hint = entity.GetContents()->GetCoverageHint();
+
+    if (coverage_hint.has_value()) {
+      auto maybe_subpass_coverage =
+          subpass_coverage.Intersection(*coverage_hint);
```

Comment:
This was the bug: Grabbing the minimum size isn't enough -- we need to intersect the rectangles and then render the blended result to the minimal subpass texture. That also means we need to offset both the geometry drawn to the subpass as well as the returned Entity's transform to account for this.